serializable/helpers.py

```diff
@@ -205,7 +205,7 @@ def deserialize(cls, o: Any) -> datetime:
 
             # Ensure any milliseconds are 6 digits
             # Background: py<3.11 supports six or less digits for milliseconds
-            v = re_sub(r'\.(\d{1,6})', lambda m: f'.{int(m.group()[1:]):06}', v)
+            v = re_sub(r'\.(\d{1,})', lambda m: f'.{int(m.group()[1:]):06}'[0:7], v)
```

Comment:
lets go with a much simpler solution: 

```py
# Ensure any milliseconds are 6 digits at max.
# Background: py<3.11 supports six or less digits for milliseconds
v = re_sub(r'\(.\d{1,6})\d*', r'\1', v)
```

Reply:
That does solve the incorrect conversion but it does not fix the decimal second conversion issue as Python 3.10 and earlier only allow 6 or 0 digits, less than 6 (unless its 0) is not allowed.

I have modified it so now it should convert it correctly and ensure either 0 or 6 digits.

Reply:
good catch!
here is a followup, that might be taken into account: https://github.com/madpah/serializable/pull/138#discussion_r1782708116